userspace/engine/falco_engine.cpp

```diff
@@ -377,24 +383,54 @@ std::unique_ptr<falco_engine::rule_result> falco_engine::process_event(std::size
 		source = find_source(source_idx);
 	}
 
-	if(should_drop_evt() || !source || !source->ruleset->run(ev, source->m_rule, ruleset_id))
+	if(should_drop_evt() || !source)
+	{
+		return nullptr;
```

Comment:
before this patch we sent `return std::unique_ptr<struct rule_result>();` now we send `nullptr` is this intentional?

Reply:
I think there's no need to construct an empty `rule_result`, `nullptr` will do what we need, and the result is handled in the `do_inspect` function